### PR TITLE
server: make long-data type no longer check nullBitmap

### LINKED ISSUE #7572: prepare insert/update long-data type cols got null using mariadb connector-c
1. What did you do?

using mariadb c client libary https://downloads.mariadb.org/connector-c/

to insert/update TEXT, TINYTEXT, MEDIUMTEXT, LONGTEXT and BLOB, TINYBLOB, MEDIUMBLOB, LONGBLOB columns.

2. What did you expect to see?

the value provided

3. What did you see instead?

always got null

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

2.0.1.rc1

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

fixes #7572

mariadb send none zero value in nullBitmap, tidb follow [protocol docs](https://dev.mysql.com/doc/internals/en/com-stmt-execute.html) to check nullBitmap and got nul value

but after experiment and check code, `nullBitmap` is nouse for `LONG_DATA_VALUE`(and it's NULL value seem be magic), so we need skip them, and make mariadb client user work.

### What is changed and how it works?

if param is long_data_value then skip nullBitmap check.

### Check List 

 - Integration test by https://github.com/pingcap/tidb-test/pull/615
 - Manual test

use mariadb client can reproduce it, after this PR fixed.

Code changes

 - impl

Side effects

 - no

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/7573)

### CODE REVIEW

**Comment by lysu** [comment]
@XuHuaiyu yes, I will do it.

File: server/conn_stmt.go [hunk 1]
```diff
@@ -246,15 +246,16 @@ func parseStmtArgs(args []interface{}, boundParams [][]byte, nullBitmap, paramTy
...
-		if nullBitmap[i>>3]&(1<<(uint(i)%8)) > 0 {
-			args[i] = nil
-			continue
-		}
 		if boundParams[i] != nil {
 			args[i] = boundParams[i]
 			continue
 		}
 
+		if nullBitmap[i>>3]&(1<<(uint(i)%8)) > 0 {
```
Comment (shenli) [inline server/conn_stmt.go]: Please add a comment for the modification.

**Comment by lysu** [comment]
@zhexuany https://github.com/pingcap/tidb-test/pull/615 integration test is enough, fake byte array ut is meanless and hard maintain.

**Comment by zhexuany** [comment]
Gotcha.

**Pull request merged**